canine/backends/dockerTransient.py

```diff
@@ -288,78 +245,14 @@ def closure():
-                shell = True
-            )
-
-        # otherwise, check that NFS is a valid node, and if so, start if necessary
-        else:
-            canine_logging.info("Found preexisting NFS server " + nfs_nodename)
-
-            # make sure NFS was created by Canine
-            if "caninetransientimage" not in nfs_inst["tags"]:
-                raise RuntimeError("Preexisting NFS server was not created by Canine.")
-
-            # make sure boot disk image matches image in config.
-            nfs_inst_details = self._pzw(gce.instances().get)(instance = nfs_nodename).execute()
-            nfs_boot_disk = [x for x in nfs_inst_details["disks"] if x["boot"]][0]
-            nfs_disk = self._pzw(gce.disks().get)(
-                         disk = re.sub(r".*/(.*)$", r"\1", nfs_boot_disk["source"])
-                       ).execute()
-            nfs_image = re.sub(r".*/(.*)$", r"\1", nfs_disk["sourceImage"])
-            if nfs_image != self.config["image"]:
-                raise RuntimeError("Preexisting NFS server's image {ni} does not match image {ci} defined in configuration.".format(ni = nfs_image, ci = self.config["image"]))
-
-            # if we passed these checks, start the NFS if necessary
-            # TODO: use the API for this
-            if nfs_inst["status"] == "TERMINATED":
-                canine_logging.print("Starting preexisting NFS server ... ", end = "", flush = True)
-                subprocess.check_call(
-                    """gcloud compute instances start {ni} --zone {z} \
-                    """.format(ni = nfs_nodename, z = nfs_inst["zone"]),
-                    shell = True
-                )
-                canine_logging.print("done", flush = True)
 
-        # start NFS monitoring thread
-        self.NFS_monitor_lock = threading.Event()
-        self.NFS_monitor_thread = threading.Thread(
-          target = self.autorestart_preempted_node,
-          args = (nfs_nodename,)
-        )
-        self.NFS_monitor_thread.start()
-
-        self.NFS_server_ready = True
-
-    def mount_NFS(self):
-        if not self.NFS_server_ready:
-            raise Exception("Need to start NFS server before attempting to mount!")
-
-        # TODO: don't hardcode this script path; make it a parameter instead
-        nfs_prov_script = os.path.join(
-                            os.path.dirname(__file__),
-                            'slurm-docker/src/nfs_provision_worker.sh'
-                          )
-        nfs_nodename = self.config["worker_prefix"] + "-nfs"
-
-        subprocess.check_call("{nps} {nnn}".format(
-          nps = nfs_prov_script, nnn = nfs_nodename
-        ), shell = True)
-
-        self.NFS_ready = True
+        ## Check if /mnt/nfs is created
+        if not os.path.exists("/mnt/nfs"):
+            subprocess.check_call("sudo mkdir /mnt/nfs", shell=True)
+            subprocess.check_call("sudo chmod 777 /mnt/nfs", shell=True)
```

Comment:
My only reservation with this whole PR is here: generally the boot disk is quite small, and thus (a) has bad IO performance and (b) will fill quickly.

I don't think we should create and attach a new disk here (though disk management would be a nice thing to add), but perhaps we can warn the user if the disk is small?

Reply:
I agree with that. I think creating/managing the disk might be done when running setup.sh at slurm_gcp_docker. There should be options to either use boot disk or mount a new/existing disk. A warning here would be nice.

Check out my new commit.